src/schemapack/_internals/exceptions.py

```diff
@@ -249,3 +249,11 @@ def __init__(self, *, class_name: str, resource_id: str):
         super().__init__(message)
         self.class_name = class_name
         self.resource_id = resource_id
+
+
+class InvalidEmbeddingProfileError(BaseError, ValueError):
+    """Raised when an embedding profile is invalid.
+    e.g. when it contains a value that is not a boolean or a nested dictionary:
+    {"some_relation": "this is not a boolean or a dictionary"}}
+    {"some_relation": {"some_relation": "this is not a boolean"}}
```

Comment:
This is a nested dictionary, but the issue is still that it's not a boolean.
Description might need to be a bit more specific on what's expected inside the dictionary.